src/storage/settings.ts

```diff
@@ -50,6 +50,33 @@ export const foodLogTotalsPositionAtom = atomWithStorage<
   getOnInit: true,
 });
 
+export const foodLogGoalsPositionAtom = atomWithStorage<
+    "hidden" | "top" | "bottom" | "both"
+>("macro-goals:position", "top", undefined, {
+    getOnInit: true,
+});
+
+export const caloriesGoalAtom = atomWithStorage<number>(
```

Comment:
It'd probably better to have a single atom with all of the goals, e.g.

```ts
interface NutritionMacroGoals {
  calories: number;
  protein: number;
  // etc
}

const DEFAULT_MACRO_GOALS = {
  calories: 2000,
  protein: 50,
  // etc
};

export const macroGoalsAtom = atomWithStorage<NutritionMacroGoals>(
  "nutrition-goals:macros",
  DEFAULT_MACRO_GOALS,
  undefined,
  { getOnInit: true }
);

export const enableNetCarbsAtom = atomWithStorage<boolean>(
  "nutrition-goals:enable-net-carbs",
  false,
  undefined,
  { getOnInit: true }
);
```

Some reasons:
- Easier to provide a reset to default for all the values, or have different sets of default values to choose from
- Less boilerplate (`atomWithStorage`, `useAtom`)
- Easier to write loops/maps that iterate over each of the values

For setting values, you'd do something like:

```ts
const [macroGoals, setMacroGoals] = useAtom(macroGoalsAtom);

// Update one goal value
const setMacroGoal = (key: keyof NutritionMacroGoals, value: number) {
  setMacroGoals({
    ...macroGoals,
    [key]: value
  });
};
```

(Note that the jotai-optics library can help do this type of scoped updating for you, but this is simple enough that it's easier to just do it manually)